File: examples/double_free.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define BUFFER_SIZE 15

int main() {
    char *buffer = malloc(BUFFER_SIZE * sizeof(char));

    printf("Enter password:\n");
    fgets(buffer, BUFFER_SIZE, stdin);

    if (strcmp(buffer, "pass123")) {
        printf("Wrong password\n");
        free(buffer);
    } else {
        printf("Correct password\n");
        free(buffer);
    }
    free(buffer);
}

File: examples/double_free.rs
#![feature(alloc_system)]
extern crate alloc_system;

use std::io::{self, BufRead};

fn main() {
    println!("Enter password:");
    let stdin = io::stdin();
    let buffer = stdin.lock().lines().next().unwrap().unwrap();

    if buffer == "pass123" {
        println!("Correct password!");
        std::mem::drop(buffer);
    } else {
        println!("Wrong password!");
        std::mem::drop(buffer);
    }
    std::mem::drop(buffer);
}
